clang/lib/DPCT/DPCT.cpp

```diff
@@ -1199,16 +1199,26 @@ int runDPCT(int argc, const char **argv) {
 
   if (MigrateBuildScriptOnly ||
       DpctGlobalInfo::getBuildScript() == BuildScriptKind::BS_Python) {
-    SmallString<128> PythonRuleFilePath(DpctInstallPath.getCanonicalPath());
-    llvm::sys::path::append(
-        PythonRuleFilePath,
-        Twine("extensions/python_rules/"
-              "python_build_script_migration_rule_ipex.yaml"));
-    if (llvm::sys::fs::exists(PythonRuleFilePath)) {
-      std::vector<clang::tooling::UnifiedPath> PythonRuleFiles{
-          PythonRuleFilePath};
-      importRules(PythonRuleFiles);
-      // generage helper functions file in the outroot dir here
+    // check if RuleFilePaths contains any user specified python migration rule
+    // file
+    bool pythonRuleFilePresent = std::any_of(
+        RuleFilePath.begin(), RuleFilePath.end(),
+        [](const clang::tooling::UnifiedPath &path) {
+          return path.getPath().contains("python_build_script_migration_rule");
+        });
+
+    if (!pythonRuleFilePresent) {
+      SmallString<128> PythonRuleFilePath(DpctInstallPath.getCanonicalPath());
+      llvm::sys::path::append(
+          PythonRuleFilePath,
+          Twine("extensions/python_rules/"
+                "python_build_script_migration_rule_pytorch.yaml"));
+      if (llvm::sys::fs::exists(PythonRuleFilePath)) {
+        std::vector<clang::tooling::UnifiedPath> PythonRuleFiles{
+            PythonRuleFilePath};
+        importRules(PythonRuleFiles);
+        // generage helper functions file in the outroot dir here
+      }
```

Comment:
1. If both two yaml files(clang/tools/dpct/DpctOptRules/python_build_script_migration_rule_ipex.yaml, clang/tools/dpct/DpctOptRules/python_build_script_migration_rule_pytorch.yaml) exisit,  which yaml files take effect?

2. clang/tools/dpct/DpctOptRules/python_build_script_migration_rule_ipex.yaml, clang/tools/dpct/DpctOptRules/python_build_script_migration_rule_pytorch.yaml have duplicate migration rules, we can keep only one copy, so that it is easy to maintain them.

Reply:
1. If any yaml file is specified with **--rule-file** option, then that rule file is chosen
    a. If no **--rule-file** options is provided, by default, **python_build_script_migration_rule_pytorch.yaml** is chosen. 

2. Shall I create another yaml file that is always loaded by default regardless of while **--rule-file** is loaded?

Reply:
In my opinion:
1. When option " --migrate-build-script=Python " or " --migrate-build-script=Python --migrate-build-script-only" is specified, then python_build_script_migration_rule_pytorch.yaml  is default loaded for pytorch xpu build env
2.   if option " --migrate-build-script=Python " or " --migrate-build-script=Python --migrate-build-script-only" is specified, and  --rule-file =  python_build_script_migration_rule_ipex.yaml is specified, then python_build_script_migration_rule_ipex.yam is loaded for ipex build env.
3. As to the common rules between python_build_script_migration_rule_pytorch.yaml and python_build_script_migration_rule_ipex.yaml should be only left in python_build_script_migration_rule_pytorch.yaml, and python_build_script_migration_rule_ipex.yaml files only keeps  the rule specific for IPEX.